src/Snak/SnakObject.php

```diff
@@ -113,7 +113,22 @@ public function serialize() {
 	 * @param string $serialized
 	 */
 	public function unserialize( $serialized ) {
-		$this->propertyId = PropertyId::newFromNumber( unserialize( $serialized ) );
+		$this->propertyId = self::newPropertyId( unserialize( $serialized ) );
+	}
+
+	/**
+	 * @param mixed $unserialized
+	 *
+	 * @return PropertyId
+	 */
+	protected static function newPropertyId( $unserialized ) {
+		if ( is_int( $unserialized ) ) {
```

Comment:
This also needs redoing with the change above, and should probably be inlined above:
```php
$numericId = unserialize( $serialized );
$this->propertyId = is_int( $numericId )
    ? PropertyId::newFromNumber( $numericId )
    : new PropertyId( $serialized );
```